sdks/java/io/jms/src/main/java/org/apache/beam/sdk/io/jms/JmsIO.java

```diff
@@ -586,7 +605,13 @@ public boolean advance() throws IOException {
       try {
         Message message;
         synchronized (this) {
-          message = this.consumer.receiveNoWait();
+          if (currentMessage != null || receiveTimeoutMillis == 0L) {
```

Comment:
If receiveTimeoutMillis is provided, is it possible to not perform the "currentMessage" check? 

Otherwise the logic appears to aggressively hit a server until it first gets a message.

Reply:
if receiveTimeoutMillis is present and there isn't a message, currentMessage is always null, so it will goes to receive with timeout, instead of receiveNoWait, is this not desirable?

Do you mean receive with timeout or receiveNoWait will "hit server"?

Reply:
Pardon my mistyping. I wrote "until it first gets a message" and meant to write "after it first gets a message". At that stage currentMessage is non-null and the code will receiveNoWait(). 

Meanwhile, receive with a recieveTimeOutMillis will always consume a message even if it arrives before the timeout. 

In a scenario where published messages are arriving less frequently, it would have a moment of quickly reaching out.

It does not appear to have a significant impact on calls but it makes it easier to triage what is happening where all calls have millisecond timeout above a threshold.

Reply:
I did this because I was wondering if `receive` will add more overload than `receiveNoWait`. In the case of lot of backlogs, it may be beneficial to have `receiveNoWait` to keep the latency as low as currently it does (for greedy polling)

Reply:
If "It does not appear to have a significant impact on calls ", I am going to simplify the logic as suggested